src/overlays/actors/ovl_En_Mnk/z_en_mnk.c

```diff
@@ -28,191 +55,2186 @@ ActorInit En_Mnk_InitVars = {
+                    this->unk_3C8 = Rand_ZeroFloat(2.0f) + 1.0f;
+                    break;
+            }
+
+            this->unk_3D4 = animIndex;
+        }
+    }
+}
+
+void func_80AB60FC(EnMnk* this, PlayState* play) {
+    f32 depthInWater;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (EnMnk_FollowPath(this, 0) != MONKEY_FOLLOWPATH_RESULT_FOLLOWING) {
+        this->pathIndex = this->path->additionalPathIndex;
+        if (this->pathIndex == 0xFF) {
+            Flags_SetSwitch(play, MONKEY_GET_SWITCHFLAG(this) + 1);
+            Actor_Kill(&this->picto.actor);
+            return;
+        }
+    }
+
+    depthInWater = this->picto.actor.depthInWater;
+    if (depthInWater > 0.0f) {
+        this->picto.actor.world.pos.y += depthInWater;
+        if ((s32)this->skelAnime.curFrame == 0) {
+            EnMnk_SpawnWaterEffects(this, play);
+        }
+        EnMnk_PlayWaterWalkSfx(this);
+        return;
+    }
+    if (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
+        EnMnk_PlayWalkSfx(this);
+    }
+}
+
+void func_80AB61E8(EnMnk* this, PlayState* play) {
+    this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0xBB8, 0xC8);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        switch (play->msgCtx.currentTextId) {
+            case 0x7D9:
+                Message_ContinueTextbox(play, 0x7DA);
+                EnMnk_Monkey_SetAnim(this, 3);
+                break;
+
+            case 0x7DA:
+                Message_ContinueTextbox(play, 0x7DB);
+                EnMnk_Monkey_SetAnim(this, 2);
+                break;
+
+            case 0x7DB:
+                Message_CloseTextbox(play);
+                EnMnk_StartNextPath(this, play);
+                this->destPointIndex = 0;
+                this->actionFunc = func_80AB60FC;
+                this->picto.actor.velocity.y = -20.0f;
+                this->picto.actor.terminalVelocity = -20.0f;
+                break;
+
+            default:
+                break;
+        }
+    }
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void func_80AB6304(EnMnk* this, PlayState* play) {
+    this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0xBB8, 0xC8);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        this->unk_3E4 |= 4;
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk_3E4 & 0x4) {
+        Message_ContinueTextbox(play, 0x7D9);
+        EnMnk_Monkey_SetAnim(this, 7);
+        this->actionFunc = func_80AB61E8;
+        this->unk_3E4 &= ~0x4;
+    }
+}
+
+void func_80AB63CC(EnMnk* this, PlayState* play) {
+    s32 pad;
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        this->unk_3E4 |= 4;
```

Comment:
```suggestion
        this->unk_3E4 |= 0x4;
```
Other uses of this are in hex, not sure if we wanted to keep it consistent.